ui/src/dashboards/components/CellEditorOverlay.tsx

```diff
@@ -497,26 +509,12 @@ class CellEditorOverlay extends Component<Props, State> {
     }
   }
 
-  private findSelectedSource = (): string => {
-    const {source} = this.props
-    const sources = this.formattedSources
-    const currentSource = getDeep<SourcesModels.Source | null>(
-      this.state.queriesWorkingDraft,
-      '0.source',
-      null
-    )
+  private get service() {
+    const {selectedService} = this.state
 
-    if (!currentSource) {
-      const defaultSource: SourcesModels.Source = sources.find(
-        s => s.id === source.id
-      )
-      return (defaultSource && defaultSource.text) || 'No sources'
+    if (selectedService) {
```

Comment:
I don't see why we need this if you're not returning anything on the else?

Reply:
good point